runtime/lua/vim/lsp/_transport.lua

```diff
@@ -0,0 +1,182 @@
+local uv = vim.uv
+local log = require('vim.lsp.log')
+
+local is_win = vim.fn.has('win32') == 1
+
+--- Checks whether a given path exists and is a directory.
+---@param filename string path to check
+---@return boolean
+local function is_dir(filename)
+  local stat = uv.fs_stat(filename)
+  return stat and stat.type == 'directory' or false
```

Comment:
Looks like a vote in favor of `vim.fs.isdir` :)

Reply:
It does overlap with `vim.fn.isdirectory`, but that function returns `0|1`, so I'd be in favour of `vim.fs.isdir`.